src/base/reconstruction.h

```diff
@@ -178,9 +181,21 @@ class Reconstruction {
   void Normalize(const double extent = 10.0, const double p0 = 0.1,
                  const double p1 = 0.9, const bool use_images = true);
 
+  // Compute the centroid of the 3D points
+  Eigen::Vector3d ComputeCentroid(double p0 = 0.1, double p1 = 0.9) const;
+
+  // Compute the bounding box corners of the 3D points
+  std::pair<Eigen::Vector3d, Eigen::Vector3d> ComputeBoundingBox(double p0 = 0.0, double p1 = 1.0) const;
+
   // Apply the 3D similarity transformation to all images and points.
   void Transform(const SimilarityTransform3& tform);
 
+  // Creates a cropped reconstruction using the input bounds as corner points
+  // of the bounding box containing the included 3D points of the new reconstruction.
+  // Only the cameras and images of the included points are registered.
+  bool Crop(const std::pair<Eigen::Vector3d, Eigen::Vector3d>& boundary,
+            Reconstruction& reconstruction) const;
```

Comment:
Not sure if getting the output by reference is better than simply returning the cropped reconstruction. Was trying to avoid unnecessary copies of large objects, but maybe it's better to change this after all.

Open to suggestions on how to change that if needed.

Reply:
You can directly return a Reconstruction object. Copy elision will ensure that there is no overhead when returning the obejct.